MapleServer2/Packets/ShopPacket.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Collections.Generic;
+using Maple2Storage.Types;
+using Maple2Storage.Types.Metadata;
+using MaplePacketLib2.Tools;
+using MapleServer2.Constants;
+
+namespace MapleServer2.Packets
+{
+    public static class ShopPacket
+    {
+        private enum ShopMode : byte
+        {
+            Open = 0x0,
+            LoadProducts = 0x1,
+            Buy = 0x4,
+            Sell = 0x5,
+            Reload = 0x6
+        }
+
+        public static Packet Open(ShopMetadata shop)
+        {
+            PacketWriter pWriter = PacketWriter.Of(SendOp.SHOP);
+            pWriter.WriteByte((byte) ShopMode.Open);
```

Comment:
Use the correct writer for Enums. this is a suggestion using it, same goes to all others mode
```suggestion
            pWriter.WriteEnum(ShopMode.Open);
```